trin-core/src/types/header.rs

```diff
@@ -61,14 +57,14 @@ where
 impl Header {
     /// Returns the Keccak-256 hash of the header.
     pub fn hash(&self) -> H256 {
-        keccak_hash::keccak(self.rlp(true))
-    }
-
-    /// Returns the RLP representation of the header.
-    pub fn rlp(&self, with_seal: bool) -> Bytes {
-        let mut s = RlpStream::new();
-        self.stream_rlp(&mut s, with_seal);
-        s.out().freeze()
+        match self.number {
+            // return hard coded mainnet genesis block hash
+            0 => H256::from_slice(
```

Comment:
@carver Am I missing something obvious here? I cannot for the life of me get the genesis block hash to match. I've seen it hardcoded in the geth codebase... Is it a hardcoded value? Is its header encoded in any special way as opposed to a regular block?

Reply:
It's not hardcoded in py-evm AFAICT. Seems like it's calculated normally from:
https://github.com/ethereum/py-evm/blob/7674e2e0caefa8b1208f650bb71a07c65ef2f305/eth/chains/mainnet/__init__.py#L140-L155